aws_jupyter_proxy/awsproxy.py

```diff
@@ -328,9 +328,12 @@ def _get_signed_canonical_headers(self) -> Tuple[str, str]:
         canonical_headers = {}
 
         for signed_header in self.upstream_auth_info.signed_headers:
-            canonical_headers[signed_header] = self.upstream_request.headers[
-                signed_header
-            ]
+            try:
+                canonical_headers[signed_header] = self.upstream_request.headers[
+                    signed_header
+                ]
+            except KeyError:
+                raise HTTPError(400, message=f"Bad Request")
```

Comment:
Can we provide more information about the failure that happened, like the fact that the `Authorization` header had expected the presence of other headers and which header was missing?

Would that be too much information or would it be a security issue to provide?

Reply:
We don't expect a keyerror unless it is intentionally malformed request so we chose to return a generic 400. We don't expect customers to malform request, so this is fine

Reply:
Who is generating the `Authorization` header attached to the request?

Reply:
Tracking improvements of better error message with header specification in https://t.corp.amazon.com/P74816551

Reply:
Discussed with Daniel on a call, this will be taken as a followup item to either:
 1. Add a log statement which captures which header was missing for the customer
 2. Update the error message to communicate which header was expected and not present